File: 1331. Rank Transform of an Array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        int[] temp = new int[arr.length];
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<arr.length;i++){
            temp[i] = arr[i];
        }
        Arrays.sort(temp);
        int count=1;
        for(int i=0;i<temp.length;i++){
            if(map.get(temp[i]) == null) {
                map.put(temp[i], count);
                count++;
            }
        }
        for(int i=0;i<arr.length;i++){
            arr[i] = map.get(arr[i]);
        }

        return arr;
    }
}

File: 1331. Rank Transform of an Array.rs
use std::collections::BTreeMap;
use std::collections::BTreeSet;

impl Solution {
    pub fn array_rank_transform(arr: Vec<i32>) -> Vec<i32> {
      let mut _arr = arr.clone();
      _arr.sort_unstable();
      _arr.dedup();
      let mut out: Vec<i32> = Vec::with_capacity(arr.len());
      // println!("{:?}", map);

      for a in &arr {
        out.push((_arr.binary_search(a).unwrap()+1) as i32);
      }

      return out;
    }
}
